scripts/python/step_2.py

```diff
@@ -1,105 +1,138 @@
-### Step 2 -- run this script locally
-### This will clone the project repo from GitHub,
-### Copy over our migration script and CI config file,
-### And push up a new branch to the repo to trigger our CI job
-### The CI job will call ci_migration_script.py
-
-from git import Repo, Actor
-from os import listdir, path, getenv, makedirs, getcwd, chdir
-from requests import get, post, patch
-from shutil import rmtree
-from shutil import copy
-from pathlib import Path as Path
-from functools import reduce
+### Step 3 -- run this script locally
+### Download the JSON file from S3 containing the project's secrets,
+### and POST these as project-level environment variables in CircleCI SaaS 
+
+### The project must already be 'Followed' and 'Building' in CircleCI SaaS
 
-# pip install gitpython
+### WARNING: If env vars with the same names already exist in the SaaS project,
+### This script will OVERWRITE them! 
+
+from requests import get, post, put, exceptions
+from os import getenv, path, getcwd
+from json import dumps, load, loads
+from boto3 import s3, client
+from botocore.exceptions import ClientError
+from jq import compile
+from github import Github
+from functools import reduce
 
 org = getenv("MIGRATION_ORG")
 project = getenv("CIRCLE_PROJECT_REPONAME")
-github_api_endpoint = 'https://api.github.com'
+tempBranch = 'get-secrets'
+gitClonePath = reduce(path.join,[getcwd(), 'git-clone', project])
 
-githubHeaders = { "Authorization": 'token {}'.format(getenv("MIGRATION_GITHUB_TOKEN")),
-                   "Accept": "application/vnd.github.v3+json" 
-                 }
+cloudV2URL = "https://circleci.com/api/v2/project/gh/{}".format(org)
+cloudV1URL = "https://circleci.com/api/v1.1/project/github/{}".format(org)
+onPremURL = "https://circleci.{}.engineering/api/v1.1/project/gh/{}".format(org, org)
```

Comment:
Break out and replace on-prem endpoint here.

Reply:
Also the /gh/ stuff too, in case other orgs are using different VCS' (which these scripts obviously do not support well, but still)